### PR TITLE
STYLE: Use itk::ReadImage and itk::WriteImage in src/Core examples

### PR DESCRIPTION
**Pull request by josetascon**
Ref #226.

### CODE REVIEW

File: src/Core/Common/AddNoiseToBinaryImage/Code.cxx [hunk 1]
```diff
@@ -63,10 +59,7 @@ main(int argc, char * argv[])
   }
 
   // Write the file
```
Comment (dzenanz) [inline src/Core/Common/AddNoiseToBinaryImage/Code.cxx:61]: As this comment now applies to only one line, and that line invokes `WriteImage`, the comment is now useless. Removal is preferred.

File: src/Core/Common/FindMaxAndMinInImage/Code.cxx [hunk 2]
```diff
@@ -46,24 +46,19 @@ main(int argc, char * argv[])
...
     return EXIT_FAILURE;
   }
-  std::string inputFilename = argv[1];
-
-  using ReaderType = itk::ImageFileReader<ImageType>;
-  ReaderType::Pointer reader = ReaderType::New();
-
-  reader->SetFileName(inputFilename.c_str());
-  reader->Update();
+  // Read image file name
```
Comment (dzenanz) [inline src/Core/Common/FindMaxAndMinInImage/Code.cxx:49]: As the following line is kind of self explanatory, this comment is not very useful.

File: src/Core/Common/ImageRegionOverlap/Code.cxx [hunk 3]
```diff
@@ -136,15 +136,10 @@ main(int argc, char * argv[])
     ++itC;
   }
 
-  // Writer
-  using FileWriterType = itk::ImageFileWriter<RGBImageType>;
-  FileWriterType::Pointer writer = FileWriterType::New();
-  writer->SetFileName(out_file_name);
-  writer->SetInput(image);
-
+  // Write image
```
Comment (dzenanz) [inline src/Core/Common/ImageRegionOverlap/Code.cxx:139]: These kinds of comments should now be mostly removed.

File: src/Core/ImageFunction/ResampleSegmentedImage/Code.cxx [hunk 4]
```diff
@@ -100,13 +97,9 @@ main(int argc, char * argv[])
...
   resizeFilter->SetInterpolator(gaussianInterpolator);
 
-  using WriterType = itk::ImageFileWriter<ImageType>;
-  WriterType::Pointer writer = WriterType::New();
-  writer->SetInput(resizeFilter->GetOutput());
-  writer->SetFileName(outputImageFileLabelImageInterpolator);
   try
   {
-    writer->Update();
+    itk::WriteImage(resizeFilter->GetOutput(), outputImageFileLabelImageInterpolator);
```
Comment (dzenanz) [inline src/Core/ImageFunction/ResampleSegmentedImage/Code.cxx:102]:
```suggestion
    itk::WriteImage(resizeFilter->GetOutput(), outputImageFileLabelImageInterpolator, true); // compress 
```
When writing label images, compressing them is quite desirable.

File: src/Core/ImageFunction/ResampleSegmentedImage/Code.cxx [hunk 5]
```diff
@@ -118,10 +111,9 @@ main(int argc, char * argv[])
   NearestNeighborInterpolatorType::Pointer nearestNeighborInterpolator = NearestNeighborInterpolatorType::New();
   resizeFilter->SetInterpolator(nearestNeighborInterpolator);
 
-  writer->SetFileName(outputImageFileNearestNeighborInterpolator);
   try
   {
-    writer->Update();
+    itk::WriteImage(resizeFilter->GetOutput(), outputImageFileNearestNeighborInterpolator);
```
Comment (dzenanz) [inline src/Core/ImageFunction/ResampleSegmentedImage/Code.cxx:116]: also compress

File: src/Core/SpatialObjects/ConvertSpatialObjectToImage/Code.cxx [hunk 6]
```diff
@@ -87,16 +87,10 @@ main(int argc, char * argv[])
...
-  WriterType::Pointer writer = WriterType::New();
-
-  writer->SetFileName(argv[1]);
-  writer->SetInput(imageFilter->GetOutput());
-
   try
   {
     imageFilter->Update();
-    writer->Update();
+    itk::WriteImage(imageFilter->GetOutput(), argv[1]);
```
Comment (dzenanz) [inline src/Core/SpatialObjects/ConvertSpatialObjectToImage/Code.cxx:93]: I think this is also a label image, so compression is desirable.

**Review by dzenanz — COMMENTED** [review]
This looks great. Minor remarks in-line.

**Comment by dzenanz** [comment]
Also, you should use clang-format to apply ITK's style. See discrepancies here:
https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/pull/303/checks?check_run_id=3000997260

File: src/Core/Common/AddNoiseToBinaryImage/Code.cxx (hunk 1 above)
Comment (josetascon) [reply src/Core/Common/AddNoiseToBinaryImage/Code.cxx]: Done

File: src/Core/Common/FindMaxAndMinInImage/Code.cxx (hunk 2 above)
Comment (josetascon) [reply src/Core/Common/FindMaxAndMinInImage/Code.cxx]: Done

File: src/Core/Common/ImageRegionOverlap/Code.cxx (hunk 3 above)
Comment (josetascon) [reply src/Core/Common/ImageRegionOverlap/Code.cxx]: Done

File: src/Core/ImageFunction/ResampleSegmentedImage/Code.cxx (hunk 4 above)
Comment (josetascon) [reply src/Core/ImageFunction/ResampleSegmentedImage/Code.cxx]: Done

File: src/Core/ImageFunction/ResampleSegmentedImage/Code.cxx (hunk 5 above)
Comment (josetascon) [reply src/Core/ImageFunction/ResampleSegmentedImage/Code.cxx]: Done

File: src/Core/SpatialObjects/ConvertSpatialObjectToImage/Code.cxx (hunk 6 above)
Comment (josetascon) [reply src/Core/SpatialObjects/ConvertSpatialObjectToImage/Code.cxx]: Done

**Comment by josetascon** [comment]
> Also, you should use clang-format to apply ITK's style. See discrepancies linter points out:
> https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/pull/303/checks?check_run_id=3000997260

Fixed.

**Comment by josetascon** [comment]
@dzenanz, I have resolved your comments and deleted unnecessary file reader and writer imports.

**Review by dzenanz — APPROVED** [review]
Awesome!

**Pull request merged**